frontend/src/utils/payment-links.ts

```diff
@@ -1,7 +1,7 @@
 import { copyToClipboard } from 'quasar';
 import { TokenInfoExtended } from 'components/models';
 import { utils as umbraUtils } from '@umbra/umbra-js';
-import useWalletStore from 'src/store/wallet';
+import { providerExport as provider, relayerExport as relayer, tokensExport as tokens } from 'src/store/wallet';
```

Comment:
_(mostly arbitrary place to stark a thread)_

Just manually tested and found this issue:
* on arbitrum, [this payment link](http://localhost:8080/send?to=vitalik.eth&token=AETH&amount=0.01) didn't populate the amount

Reply:
Ah interesting, seems it only populates on mainnet. The token in the link is AETH, I wonder if the difference in naming native tokens has to do with it. Going to test using ERC20 tokens too, let's see...

Reply:
Ok here's (part of?) the bug, haven't implemented a fix yet:

In `AccountSend.vue`, the payment link is properly parsed and the component's state is properly. This happens immediately when the component mounts. In that component, we watch various state and update accordingly:

https://github.com/ScopeLift/umbra-protocol/blob/533633f7971695afe4b224ec5bef26780780ec5e/frontend/src/pages/AccountSend.vue#L333-L342

Every time any of that state changes, we verify that the selected token exists on the selected network, and if not we clear the token and amount selection:

https://github.com/ScopeLift/umbra-protocol/blob/533633f7971695afe4b224ec5bef26780780ec5e/frontend/src/pages/AccountSend.vue#L366-L373

This is the source of a bug: The data populates, then the relayer hasn't finished loading so the only token in the list is ETH, therefore our tokens are not found and we reset the token selection and amount. This line of code was required when we didn't refresh the page on network change. Now that we do that, I'm pretty sure this block can be deleted. From a quick test, deleting it caused some errors, so need to look a bit more but just documenting this for now

Reply:
I also pushed 4aa2b44 which renames all native-ETH tokens to just ETH, instead of AETH for arbitrum and OETH for optimism. When L2s were newer they tried that naming convention to help users distinguish. Now, looking at the official arbitrum and optimism bridges, it seems they no longer make that distinction.

Reply:
Just pushed 9e88061 to make payment link parsing more robust. @davidlaprade mind testing that out to confirm it fixes the issue?

Reply:
Looks great! Thanks for the fix

This is unrelated to this PR -- more like a feature request than anything -- but it'd be nice if a payment link could prompt a user to change networks, if necessary. If you're requesting payment from someone with a link, network matters. I don't want to receive (say) DAI on Polygon if I don't have Matic there. Currently, someone could be on mainnet, create the payment link to request DAI, then the recipient could be logged in on Polygon, and paste the link thinking it's fine to send on Polygon when actually it's not.

Reply:
I was thinking about that too as I was fixing this, tracking in https://github.com/ScopeLift/umbra-protocol/issues/439